pkg/users/account-details.js

```diff
@@ -0,0 +1,428 @@
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React from 'react';
+import moment from "moment";
+import { superuser } from "superuser.jsx";
+
+import { Button } from '@patternfly/react-core';
+import { parse_passwd_content, parse_group_content } from "./parsers.js";
+import { show_unexpected_error } from "./dialog-utils.js";
+import { delete_account_dialog } from "./delete-account-dialog.js";
+import { account_expiration_dialog, password_expiration_dialog } from "./expiration-dialogs.js";
+import { set_password_dialog, reset_password_dialog } from "./password-dialogs.js";
+import { AccountRoles } from "./account-roles.js";
+import { AuthorizedKeys } from "./authorized-keys-panel.js";
+
+const _ = cockpit.gettext;
+
+function log_unexpected_error(error) {
+    console.warn("Unexpected error", error);
+}
+
+export class AccountDetails extends React.Component {
+    constructor() {
+        super();
+        this.state_user = null;
+        this.state = {
+            accounts: null,
+            groups: null,
+            userinfo: { },
+
+            currently_logged_in: false,
+            last_login_time: null,
+            locked: false,
+
+            edited_real_name: null,
+            committing_real_name: false,
+            edited_locked: null
+        };
+    }
+
+    componentDidMount() {
+        cockpit.user().then(user => this.setState({ userinfo: user }));
+
+        this.handle_passwd = cockpit.file('/etc/passwd');
+        this.handle_passwd.watch(content => {
+            this.setState({ accounts: parse_passwd_content(content) });
+            this.get_locked();
+            this.get_expire();
+        });
+
+        this.handle_shadow = cockpit.file('/etc/shadow', { superuser: "try" });
+        this.handle_shadow.watch(content => {
+            this.get_expire();
+        });
+
+        this.handle_groups = cockpit.file('/etc/group');
+        this.handle_groups.watch(content => {
+            this.setState({ groups: parse_group_content(content) });
+        });
+
+        this.componentDidUpdate({});
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.user != this.props.user) {
+            // Reset per-account state and fetch it for the new user
+            this.setState({
+                currently_logged_in: false,
+                last_login_time: null,
+                locked: false
+            });
+
+            this.get_logged();
+            this.get_locked();
+            this.get_expire();
+        }
+    }
+
+    componentWillUnmount() {
+        this.handle_passwd.close();
```

Comment:
handle_groups.close()

Reply:
Obsolete with the rewrite to use hooks.  (This was a bug that hooks are intended to prevent, and they did. :-)